### DISCUSSION TITLE
Download Sentinel2 L2A via FORCE

### DISCUSSION THREAD

**Discussion by JonasardoLundUniv** [davidfrantz/force#259] [General]
Seem to work to download L2A data using force-level1-csd by just replacing the L1C metadata file with the L2A metadata file 
(https://storage.googleapis.com/gcp-public-data-sentinel-2/L2/index.csv.gz). 
May be useful as it is faster than other services.

Regards and thanks for providing nice software.
/Jonas

[jonasardo@aurora-nateko FORCE]$ force-suite force-level1-csd -c 0,100 -d 20220101,20221201 -s S2A,S2B -k ./metadataL2A/ ./tmp/ tmplista.txt T28PDC

Querying the metadata catalogue for Sentinel2 data
Sensor(s): S2A,S2B
Tile(s): T28PDC
Daterange: 20220101 to 20221201
Cloud cover minimum: 0%, maximum: 100%

138 Sentinel2 Level 1 scenes matching criteria found
95.39GB data volume found.

Starting to download 138 Sentinel2 Level 1 scenes

Downloading S2B_MSIL2A_20221130T112329_N0400_R037_T28PDC_20221130T141952.SAFE(138 of 138)...

|==============================================================================================================================================================================| 100 % 
Done.

**Comment by ernstste** [comment]
Hi Jonas,

yes, the metadata catalogues for Level 1 and Level 2 use the same structure. So while untested, this should work just fine.
FORCE is usually used for it's processing starting with Level 1 data, so I never considered adding L2 download capabilities. 
I don't assume that people would install FORCE on their systems specifically to download S2 L2 data. If used for FORCE, the Level 2 data from GCS needs to be cubed to be used in further processing steps. 

Stefan

**Comment by JonasardoLundUniv** [comment]
Hi Stefan,

On 2022-12-22 12:39, Stefan Ernst wrote:
>
> Hi Jonas,
>
> yes, the metadata catalogues for Level 1 and Level 2 use the same 
> structure. So while untested, this should work just fine.
>
I am aware of this but currently involved in a project where we must use 
SEN2Cor processed L2A data as standard and I like the other skills of 
FORCE.
>
> FORCE is usually used for it's processing starting with Level 1 data, 
> so I never considered adding L2 download capabilities.
> I don't assume that people would install FORCE on their systems 
> specifically to download S2 L2 data. If used for FORCE, the Level 2 
> data from GCS needs to be cubed to be used in further processing steps.
>
I don't assume that either, but smooth  when already installed and I 
guess faster than the  ESA hub?
Can the L2A data be cubed with  force-cube ?


/Jonas

**Comment by ernstste** [comment]
Yes, the download should be much faster than the data hub. On top of that, you don't have to order older scenes from tape storage, which is what really used to limit the downloads from the hub in the past.

I haven't tested force-cube for the L2A data but I don't see any reason why it wouldn't work. So if your project requires S2 data processed with SEN2Cor, going this route could be a viable option.

Stefan
